pkg/filesystem/local_directory_windows.go

```diff
@@ -0,0 +1,261 @@
+// +build windows
+package filesystem
+
+import (
+	"errors"
+	"internal/syscall/windows"
+	"log"
+	"path"
+	"path/filepath"
+	"os"
+	"sort"
+	"strings"
+	"syscall"
+	"time"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+const (
+	// Windows does not differentiate permissions for users, group and others.
+	// Therefore, a mapping table is needed.
+	FileMode644 = 0666
+	FileMode666 = 0666
+	FileMode700 = 0777
+	DirMode700 = 0777
+	DirMode777 = 0777
+	SymLinkMode777 = 0666
+)
+
+const uncPrefix = `\\?\`
+
+type localDirectory struct {
+	longAbsPath string
+}
+
+func validateFilename(name string) error {
+	if name == "" || name == "." || name == ".." || strings.ContainsRune(name, '/') {
+		return status.Errorf(codes.InvalidArgument, "Invalid filename: %#v", name)
+	}
+	return nil
+}
+
+func PortableSyscallUmask(mask int) () {
+	// No implementation on Windows
+}
+
+// NewLocalDirectory creates a directory handle that corresponds to a
+// local path on the system.
+func NewLocalDirectory(path string) (Directory, error) {
```

Comment:
So the idea behind `NewLocalDirectory()` is that it's this one-time 'hatch' for converting a local directory to something managed by the Filesystem layer. What you pass in here is effectively machine-dependent. This is why I'm not fully convinced you need all of the logic below.

Why can't we simply say: on Windows, NewLocalDirectory() **MUST** be called with a UNC path?

Reply:
There is a need for a helper function somewhere, so I thought it was more convenient to keep all UNC logic inside this file than spreading it out.

(By the way, I should rename some of the `longAbsPath` to `UNC`.)